### DISCUSSION TITLE
Simulating the energy loss per turn in consistent way

### DISCUSSION THREAD

**Discussion by lnadolski** [atcollab/at#629] [General]
Dear All,

I would like to simulate the impact of energy loss per turn on the beam dynamics in a consistent way in AT.
The net energy loss per turn is the contribution of 

- radiation for magnets, especially dipoles
- radiation of ID if modeled as a series of alternating dipoles, a wiggler element, etc.
- energy loss from resistive wall 
- energy loss in the Cu wall of a **passive** harmonic cavity for instance

For SOLEIL that total energy loss can by twice the energy loss U0 for a bare lattice.

Discussing with colleagues, we could add a fake element at the end of each turn to add some extra energy loss (a marker with energy shift in T1(5) will do the job).

This help to better simulate the longitudinal transverse dynamic as shown in the following picture.
But damping times are not affected by such a trick.
The good point is that my RF bucket and RF voltage are more consistent with the total energy loss by turn.

![image](https://github.com/atcollab/at/assets/3135838/0381a665-1c4a-4c7b-9645-a38e96803065)

Any ideas or suggestions would be welcome.

Laurent.

**Comment by lfarv** [comment]
Hello @lnadolski 
What you need is slightly more complicated than just a step on T1[5]:

- first you must scale the momentum step with (1 + &delta;)<sup>2</sup>. I'm not sure of the exponent, but that's what is done in all `*RadPass` methods.
- 2nd, you must also scale p<sub>x</sub> and p<sub>y</sub> so that the angles  p<sub>x</sub>/(1 + &delta;) are kept constant. With this, the damping times will be correct.

This gives something like this (C style):
```C
   /* calculate angles from momentums 	*/
   p_norm = 1/(1+r[4]);
   xpr = r[1]*p_norm;
   ypr = r[3]*p_norm;

  /* momentum step */
   ddp = K * SQR(1+r[4]).    /• K: nominal momentum step */
   r[4] -= ddp;
   
   /* recalculate momentums from angles after losing energy for radiation 	*/
   p_norm = 1/(1+r[4]);
   r[1] = xpr/p_norm;
   r[3] = ypr/p_norm;
```
To be checked, I did not try !

I think that with this, the computation of damping times and emittances are correct. There will be no diffusion associated with this element, which makes sense only if the loss occurs in a non-dispersive section. Adding diffusion terms for wigglers and other radiating elements like this one has been already discussed, it's more complicated.

It's easy to provide a new passmethod for that. To have the `disable_6d/enable_6d` functions work, we also would like a new element class. If so, we should choose the attribute defining the element: energy loss in eV ?

I can look at that if you think it's an interesting development!

**Reply by lnadolski** [reply]
Dear @lfarv,

I think this is a useful development if you can pursue it.

Thank you for your feedback.
Rescaling px and py with the new energy makes a lot of sense indeed.
I will follow your advice and make to test.

**Comment by lnadolski** [comment]
Dear @lfarv 

I am not sure to understand the (1+delta)^2 factor for the energy.
Could you provide me with a passmethod to do the tracking?

Laurent.

**Reply by lfarv** [reply]
For the energy dependence of the radiated power, my 1st idea would have been a power 3, but I checked all the RadPass methods, it's everywhere power 2. I'll set power 2 and think about it later…

I'll 1st propose a passmethod today, to be used with a Marker. The element class and full integration will come later.

**Comment by lnadolski** [comment]
Ok, thanks.
According to Sands
<img width="827" alt="image" src="https://github.com/atcollab/at/assets/3135838/f46eb72a-fcbf-4f6e-8f7d-4ec22af58d17">

<img width="787" alt="image" src="https://github.com/atcollab/at/assets/3135838/e1f7e75b-ae99-4d54-9b13-372ec5fcfa0f">
